lightning/src/ln/functional_test_utils.rs

```diff
@@ -1813,7 +1886,8 @@ pub fn fail_payment_along_route<'a, 'b, 'c>(origin_node: &Node<'a, 'b, 'c>, expe
 		assert_eq!(path.last().unwrap().node.get_our_node_id(), expected_paths[0].last().unwrap().node.get_our_node_id());
 	}
 	expected_paths[0].last().unwrap().node.fail_htlc_backwards(&our_payment_hash);
-	expect_pending_htlcs_forwardable!(expected_paths[0].last().unwrap());
+	let expected_destinations: Vec<HTLCDestination> = iter::repeat(HTLCDestination::FailedPayment { payment_hash: our_payment_hash }).take(expected_paths.len()).collect();
+	expect_pending_htlcs_forwardable_and_htlc_handling_failed!(expected_paths[0].last().unwrap(), HTLCHandlingFailedConditions::new().htlc_processing_failed_with_reasons(expected_destinations.clone()));
```

Comment:
without the clone, the compiler complains:

```
error[E0382]: use of moved value: `expected_destinations`
    --> lightning/src/ln/functional_test_utils.rs:1890:168
     |
1889 | ...et expected_destinations: Vec<HTLCDestination> = iter::repeat(HTLCDestination::FailedPayment { payment_hash: our_payment_hash }).take(expected_paths.len()).collect();
     |       --------------------- move occurs because `expected_destinations` has type `std::vec::Vec<HTLCDestination>`, which does not implement the `Copy` trait
1890 | ...xpect_pending_htlcs_forwardable_and_htlc_handling_failed!(expected_paths[0].last().unwrap(), HTLCHandlingFailedConditions::new().htlc_processing_failed_with_reasons(expected_destinations...
     |                                                                                                                                                                         ^^^^^^^^^^^^^^^^^^^^^
     |                                                                                                                                                                         |
     |                                                                                                                                                                         value moved here
     |                                                                                                                                                                         value used here after move
```

same for the comment below. I'm not sure what's moving it, but I applied the "let's stop Rust from complaining" approach here. I tried looking into the macros/`HTLCHandlingFailedConditions` initializers downstream but couldn't find anything. But I'm new to macros, so I may have missed something. ideas? 😅